src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -896,6 +899,52 @@ class spell_dru_omen_of_clarity : public AuraScript
     }
 };
 
+// 392303 - Power of the Archdruid
+class spell_dru_power_of_the_archdruid : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellEffect({ { SPELL_DRUID_POWER_OF_THE_ARCHDRUID, EFFECT_0 } });
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* target = GetTarget();
+        Unit* procTarget = eventInfo.GetActionTarget();
+
+        // Note: range is EFFECT_0's BasePoints.
+        float const spellRange = aurEff->GetAmount();
+
+        std::list<Unit*> targetList;
+        Trinity::AnyFriendlyUnitInObjectRangeCheck checker(procTarget, procTarget, spellRange);
+        Trinity::UnitListSearcher<Trinity::AnyFriendlyUnitInObjectRangeCheck> searcher(procTarget, targetList, checker);
+        Cell::VisitAllObjects(procTarget, searcher, spellRange);
+
+        if (targetList.empty())
+            return;
+
+        AuraEffect* const powerOfTheArchdruidEffect = target->GetAuraEffect(SPELL_DRUID_POWER_OF_THE_ARCHDRUID, EFFECT_0);
+        if (!powerOfTheArchdruidEffect)
+            return;
+
+        // Note: max. targets is SPELL_DRUID_POWER_OF_THE_ARCHDRUID's EFFECT_0 BasePoints.
+        uint8 const maxTargets = powerOfTheArchdruidEffect->GetAmount();
+
+        if (targetList.size() > maxTargets)
+            Trinity::Containers::RandomResize(targetList, maxTargets);
+
+        for (Unit* chosenTarget : targetList)
+            target->CastSpell(chosenTarget, eventInfo.GetProcSpell()->GetSpellInfo()->Id, CastSpellExtraArgs(aurEff));
+
+        target->RemoveAurasDueToSpell(GetSpellInfo()->Id);
```

Comment:
You can actually call Remove() or Remove(removeMode), you dont even need to say: target->Remove(), because the aura is applied on the target anyway

`enum AuraRemoveMode
{
    AURA_REMOVE_NONE = 0,
    AURA_REMOVE_BY_DEFAULT = 1,       // scripted remove, remove by stack with aura with different ids and sc aura remove
    AURA_REMOVE_BY_INTERRUPT,         // removed by aura interrupt flag
    AURA_REMOVE_BY_CANCEL,
    AURA_REMOVE_BY_ENEMY_SPELL,       // dispel and absorb aura destroy
    AURA_REMOVE_BY_EXPIRE,            // aura duration has ended
    AURA_REMOVE_BY_DEATH
};`